terraform/eks/main.tf

```diff
@@ -111,16 +111,63 @@ resource "kubernetes_service_account" "aoc-role" {
   automount_service_account_token = true
 }
 
+resource "aws_iam_role" "STSAssumeIAMRole" {
+  name = "aoc-eks-assume-role-${module.common.testing_id}"
+
+  # Terraform's "jsonencode" function converts a
+  # Terraform expression result to valid JSON syntax.
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Action = "sts:AssumeRoleWithWebIdentity"
+        Effect = "Allow"
+        Principal = {
+          Federated = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:oidc-provider/${trimprefix(data.aws_eks_cluster.testing_cluster.identity[0].oidc[0].issuer, "https://")}"
+        }
+        Condition = {
+          StringEquals = {
+            "${trimprefix(data.aws_eks_cluster.testing_cluster.identity[0].oidc[0].issuer, "https://")}:aud" : "sts.amazonaws.com"
+          }
+        }
+      }
+    ]
+  })
+
+  inline_policy {
+    name = "aoc-inline-policy-${module.common.testing_id}"
+
+    policy = jsonencode({
+      Version = "2012-10-17"
+      Statement = [
+        {
+          Action = [
+            "ec2:*",
+            "logs:*",
+            "cloudwatch:PutMetricData",
+            "s3:*",
+            "ssm:*",
+            "xray:*"
```

Comment:
Are these the final state of the policy or will end up going back and specify them out?

Reply:
We can specify them out later. I think this should be fine because we only create the policy for the test then we tear it down after test execution so a wild card policy is not floating around.

Reply:
Considering we use a role with an admin policy currently this is really stripped down :) aoc-integ-test-dev